spm/spm_internal.h

```diff
@@ -122,15 +116,25 @@ typedef struct spm_rot_service {
  * Structure containing Partition->RoT-Service channel information.
  */
 typedef struct spm_ipc_channel {
-    spm_partition_t *src_partition; /* Pointer to the Partition which connects to the Root of Trust Service.*/
+    struct spm_partition *src_partition; /* Pointer to the Partition which connects to the Root of Trust Service.*/
     spm_rot_service_t *dst_rot_service; /* Pointer to the connected Root of Trust Service.*/
     void *rhandle; /* Reverse handle to be used for this channel.*/
     void *msg_ptr; /* message data sent from user */
     struct spm_ipc_channel *next; /* Next channel in the chain  */
-    SpmMsgType msg_type; /* The message type, one of ::spm_msg_type.*/
-    ChannelState state; /* The current processing state of the channel.*/
+    uint8_t msg_type; /* The message type.*/
+    uint8_t state; /* The current processing state of the channel.*/
```

Comment:
+1 for documenting members. We should use doxygen format (in this case `/**< text */`)

Reply:
We prefer not to generate doxygen for internal structs which are not part of the API.
Thus removed the second asterisk.
What are the best practices for mbed-os regarding doxygen? Should we generate it for all headers or just the APIs?

Reply:
I've missed internal suffix in the file name !

`/internal` should do the trick